lib/adagrams.rb

```diff
@@ -0,0 +1,90 @@
+require "awesome_print"
+
+def draw_letters
+  avail_letters = %w[A A A A A A A A A B B C C D D D D E E E E E E E E E E E E F F G G G H H I I I I I I I I I J K L L L L M M N N N N N N O O O O O O O O P P Q R R R R R R S S S S T T T T T T U U U U V V W W X Y Y Z]
+  avail_letters.sample(10)
+end
+
+sample = draw_letters
+# print sample
+
+def uses_available_letters?(input, letters_in_hand)
+  letters_in_hand = letters_in_hand.clone
+  input = input.upcase
+  input.each_char.all? do |char|
+    if letters_in_hand.include?(char)
+      letters_in_hand.delete_at(letters_in_hand.index(char))
+      true
+    else
+      false
+    end
+  end
+end
+
+def score_word(word)
+  score = []
+  score_chart = [%w[A E I O U L N R S T], %w[D G], %w[B C M P], %w[F H V W Y], %w[K]]
+  word = word.upcase
+
+  score_chart.each_with_index do |letter_bank, letter_value|
+    word.each_char do |char|
+      if letter_bank.include?(char)
+        score << (letter_value + 1)
```

Comment:
Ha! Interesting relationship between the index of the `letter_bank` inside of `score_chart` and its point value. Because this relationship isn't very trustworthy, I might suggest using a different data structure with a more obvious relationship between letter and point value, but I applaud this Sherlock Holmes sleuthing ;)